web/profiles/joinup/src/Plugin/Block/FrontpageBlock.php

```diff
@@ -0,0 +1,188 @@
+   */
+  protected $currentUser;
+
+  /**
+   * Drupal\og\MembershipManager definition.
+   *
+   * @var \Drupal\og\MembershipManager
+   */
+  protected $ogMembershipManager;
+
+  /**
+   * Drupal\Core\Entity\EntityTypeManagerInterface definition.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs a new FrontpageBlock object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param string $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Session\AccountProxy $current_user
+   *   The current user proxy.
+   * @param \Drupal\og\MembershipManager $og_membership_manager
+   *   The og membership manager service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, AccountProxy $current_user, MembershipManager $og_membership_manager, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->currentUser = $current_user;
+    $this->ogMembershipManager = $og_membership_manager;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('current_user'),
+      $container->get('og.membership_manager'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * Provides empty homepage.
+   *
+   * @return array
+   *   A render array for the homepage.
+   */
+  public function build() {
+    $groups = $this->ogMembershipManager->getUserGroups($this->currentUser->getAccount());
+    $rows = [];
+    if (!empty($groups['rdf_entity'])) {
+      $rows = $this->getContentFromMemberships($groups);
+    }
+    // @todo: Else: Provide content with site-wide content.
+    // @see: https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-3427
+
+    $build['#attributes'] = [
+      'class' => ['listing', 'listing--grid', 'mdl-grid'],
+    ];
+
+    $build += $rows;
+    // @todo: Set proper caching.
+    $build['#cache']['max-age'] = 0;
+    return $build;
+  }
+
+  /**
+   * Receives the content of the groups the user is a member of.
+   *
+   * @param array $groups
+   *   The user's memberships.
+   *
+   * @return array
+   *   An array of rows to render.
+   */
+  protected function getContentFromMemberships(array $groups) {
+    $rdf_entities = isset($groups['rdf_entity']) ? $groups['rdf_entity'] : [];
+    $cids = array_map(function ($rdf_entity) {
```

Comment:
As discussed yesterday, for some special cases we will be hitting the query size limit. Take only the first 100 groups to avoid breaking for our test roles, that have hundreds of groups. Do not set any message (see user profile).